src/Grid/grid_generators.jl

```diff
@@ -85,16 +86,19 @@ end
 
 function _generate_2d_nodes!(nodes, nx, ny, LL, LR, UR, UL)
       for i in 0:ny-1
-        ratio_bounds = i / (ny-1)
+        T = typeof(LL[1])
```

Comment:
Regarding also to grids, I don't know if this is an issue or not but when I specified my tensors (Left and Right) to be Float32 and generated grid as follows:
```julia
left = Tensor{1,2,Float32}((0,-0)) # define the left bottom corner of the grid.
right = Tensor{1,2,Float32}((100.0,100.0)) # define the right top corner of the grid.
grid = generate_grid(Quadrilateral, (100, 100),left,right);
```
and due to ` _generate_2d_nodes!`  uses Float division the by default division generates Float64 type an throws an error when tries to push a node in nodes list `push!(nodes, Node((x, y)))`
 because the former is of type Float32 whereas the latter is of type Float64. So, as a quick workaround a I did an explicit cast for `ratio_bounds` to ensure type comptability.

Reply:
This is nasty. We should be able to circumvent the issue with the following patch for now:

```
diff --git a/src/Grid/grid_generators.jl b/src/Grid/grid_generators.jl
index 42eca02dc..6e9952b58 100644
--- a/src/Grid/grid_generators.jl
+++ b/src/Grid/grid_generators.jl
@@ -69,7 +69,7 @@ end
 
 function _generate_2d_nodes!(nodes, nx, ny, LL, LR, UR, UL)
       for i in 0:ny-1
-        ratio_bounds = i / (ny-1)
+        ratio_bounds = convert(eltype(LL), (i / (ny-1)))
 
         x0 = LL[1] * (1 - ratio_bounds) + ratio_bounds * UL[1]
         x1 = LR[1] * (1 - ratio_bounds) + ratio_bounds * UR[1]
@@ -78,7 +78,7 @@ function _generate_2d_nodes!(nodes, nx, ny, LL, LR, UR, UL)
         y1 = LR[2] * (1 - ratio_bounds) + ratio_bounds * UR[2]
 
         for j in 0:nx-1
-            ratio = j / (nx-1)
+            ratio = convert(eltype(LL), j / (nx-1))
             x = x0 * (1 - ratio) + ratio * x1
             y = y0 * (1 - ratio) + ratio * y1
             push!(nodes, Node((x, y)))
```

Reply:
Can you go ahead and open a PR with the patch for master together with some test coverage for all grid generators?

Reply:
Resolved in https://github.com/Ferrite-FEM/Ferrite.jl/pull/973 .